apps/server/src/apps/helpers/request-logger-middleware.ts

```diff
@@ -0,0 +1,28 @@
+import { Request, Response, NextFunction } from 'express';
+import { Logger } from '@src/core/logger';
+import { RequestLoggable } from '@src/apps/helpers/request-loggable';
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+
+export const createRequestLoggerMiddleware = (
+	logger: Logger
+): ((request: Request, response: Response, next: NextFunction) => void) => {
+	logger.setContext('AppLoggerMiddleware');
+	const enabled = Configuration.get('REQUEST_LOGGING_ENABLED') as boolean;
+
+	return (request: Request, response: Response, next: NextFunction): void => {
```

Comment:
Der Request Logger im legacy sollte nur logs produzieren wenn sie über eine bestimmte Zeit gehen. Dort gibt es aber auch kein Interceptor der den Request unterbricht nach einer Zeit von x. Wir haben die Timeouts dann auch im legacy-client und im Server kann dann schon sowas auftauchen wie ..ich habe jetzt 30sec gebraucht. Aber der Client hat den Nutzer schon längst ein Timeout Error geworfen.

Reply:
goal is to log all requests, similar to an access log

nothing about timeouts or anything